transhumandesign/kag-base
# [fixed] No materials on next map

## Status

- **READY**

## Description
Resolves #1344 & closes PR #1526

This PR decisively fixes the bug where players would spawn on next map without receiving their materials.

Why did the bug occur?
When going next map, players would be auto-balanced, and will change teams. Changing a player's team causes the player to request a new spawn. This kills their blob and spawns a new one. However, players receive their spawn mats before the second spawn happens, and they end up dropping their materials when they die.

TLDR: Players changing teams on next map causes them to drop their mats.

## Steps to Test or Reproduce

a) Go onto a dedicated server.
b) Spawn at least 3 bots with ``!bot``. You will probably need moderator/rcon access to use the chat command.
c) Keep going next map. If you are testing *without* the applied fix you will notice that you will receive no mats every time you switch teams by auto balance. Otherwise, with the applied fix you will receive mats as usual.
